tests/test_axis_limits.py

```diff
@@ -1,95 +1,323 @@
+                "atol": custom_atol1,
+                "desired_at_axis": custom_desired_at_axis1
+            },
+            "name2": {"rtol": custom_rtol2},
+            ...
+        }
+
+    """
+    if kwargs is None:
+        kwargs = {}
+    if isinstance(names, str):
+        names = [names]
+    # TODO: remove when boozer transform works with multiple surfaces
+    names = [x for x in names if not ("Boozer" in x or "_mn" in x or x == "B modes")]
+
+    rho = np.linspace(0, 1, 10) * delta
+    grid = LinearGrid(rho=rho, M=5, N=5, NFP=eq.NFP, sym=eq.sym)
+    assert grid.axis.size
+    integrate = surface_integrals_map(grid)
+    data = eq.compute(names, grid=grid)
+
+    for name in names:
+        if _skip_this(eq, name) or data_index[name]["coordinates"] == "":
+            continue
+        # make single variable function of rho
+        if data_index[name]["coordinates"] == "r":
+            profile = data[name]
+        else:
+            # Norms and integrals are continuous functions, so their composition
+            # cannot disrupt existing continuity. Note that the absolute value
+            # before the integration ensures that a discontinuous integrand does
+            # not become continuous once integrated.
+            profile = integrate(np.abs(data[name]))
+            # integration replaced nans with 0, put them back
+            profile = np.where(np.isnan(data[name]), np.nan, profile)
+        profile = grid.compress(profile)
+        fit = kwargs.get(name, {}).get("desired_at_axis", desired_at_axis)
+        if fit is None:
+            if np.ndim(data_index[name]["dim"]):
+                # can't polyfit tensor arrays like grad(B)
+                fit = (profile[0] + profile[1]) / 2
+            else:
+                # fit the data to a polynomial to extrapolate to axis
+                poly = np.polyfit(rho[1:], profile[1:], 6)
+                # constant term is the same as evaluating polynomial at rho=0
+                fit = poly[-1]
+        np.testing.assert_allclose(
+            actual=profile[0],
+            desired=fit,
+            rtol=kwargs.get(name, {}).get("rtol", rtol),
+            atol=kwargs.get(name, {}).get("atol", atol),
+            equal_nan=False,
+            err_msg=name,
+        )
+
+
+def get_matches(fun, pattern):
+    """Return all matches of ``pattern`` in source code of function ``fun``."""
+    src = inspect.getsource(fun)
+    # attempt to remove any decorator functions
+    # (currently works without this filter, but better to be defensive)
+    src = src.partition("def ")[2]
+    # attempt to remove comments
+    src = "\n".join(line.partition("#")[0] for line in src.splitlines())
+    matches = pattern.findall(src)
+    matches = {s.replace("'", "").replace('"', "") for s in matches}
+    return matches
+
 
 class TestAxisLimits:
     """Tests for compute functions evaluated at limits."""
 
+    @pytest.mark.unit
+    def test_data_index_deps(self):
```

Comment:
This file is included in this PR, so that reviewers can skip checking what this function tests during their review.